design/oss_suite_release.md

```diff
@@ -0,0 +1,300 @@
+command line interface (CLI), client libraries, and integrations with platforms and DevOps tools.
+
+At current, we have no formal process for releasing the full suite of tools in
+the Conjur OSS ecosystem so that:
+- It is clear to end users which version of Conjur OSS they should use, and what
+  integrations and tooling are compatible with that version.
+- Key use cases that involve more than two components working together are automatically
+  tested to ensure that the end-to-end experience in those use cases works as expected.
+- Development can continue on the master branch of individual components while
+  end users have a reliable way to pull "stable" releases for all components and
+  avoid unexpectedly running "edge" releases .
+- It is clear what features are officially released and which are still in progress -
+  at current, there is no way to easily track unreleased features in our broad suite
+  of tools and integrations, and formally creating a system for tracking unreleased
+  features will help us to do a better job of comprehensively testing and releasing
+  completed features.
+
+In this effort, we will build a formal Conjur OSS Suite Release process to begin
+addressing these concerns.
+
+## Key Characteristics of Conjur OSS Suite Release
+- All key components of the Conjur OSS Suite will be grouped into a single
+  Conjur OSS Suite Release
+- There is a special [release repository](https://github.com/cyberark/conjur-oss-suite-release)
+  that is used to:
+  - Pin the versions of the components included in the suite release
+  - Automatically run test suite against versions pinned in the suite release daily
+  - When the pinned component versions are changed and the suite version is
+    incremented:
+    - Automatically generate GitHub releases that include release notes with links
+      to artifacts
+    - Automatically generate a documentation-friendly release page that includes
+      a table of contents, release notes for each component, and links to artifacts.
+    - Automatically update our [homebrew](https://github.com/cyberark/homebrew-tools)
+      and [helm charts](https://github.com/cyberark/helm-charts)
+  - Document how end users should utilize the Conjur OSS Suite Release
+
+In addition, as part of building this release, we'll also be creating:
+- A [GitHub landing page](https://cyberark.github.io/conjur) that links to the
+  repos included in the Conjur OSS Suite and describes what each one does.
+- A template to be added to the READMEs of repos in the Conjur OSS Suite to clarify
+  about the state of the code on master ("edge") and to encourage end users to
+  visit the Conjur OSS Suite Release to determine which versions of each product
+  to use.
+- A tool for visualizing the unreleased changes in the components of the Conjur OSS
+  Suite Release, to help us in understanding whether a new OSS Suite Release is
+  needed
+
```

Comment:
Can you add a "Definition of Done" header, probably here, just so that we don't forget to add this content?

Reply:
as discussed, I'll leave this off the doc and add it to the epic later - I'd like to define the DoD once this doc is approved.